Zend/zend_API.c

```diff
@@ -1144,7 +1145,23 @@ ZEND_API int zend_update_class_constants(zend_class_entry *class_type) /* {{{ */
 						}
 						ZVAL_DEREF(val);
 						if (Z_CONSTANT_P(val)) {
-							if (UNEXPECTED(zval_update_constant_ex(val, ce) != SUCCESS)) {
+							if (prop_info->type) {
+								zval tmp, tmp2, *v;
+
+								ZVAL_COPY(&tmp, val);
+								if (UNEXPECTED(zval_update_constant_ex(&tmp, ce) != SUCCESS)) {
+									return FAILURE;
+								}
+								v = zend_verify_property_type(prop_info, &tmp, &tmp2, ZEND_CALL_USES_STRICT_TYPES(EG(current_execute_data)));
```

Comment:
Does this mean that constant expression property initializers will use strict_types depending on whether the file where the object was first used has strict_types? Shouldn't this instead use the strict_types of the file the class is declared in?